src/Migration/SessionMigrationService.php

```
@@ -0,0 +1,283 @@
+     * 4. Optionally deletes the old session
+     *
+     * IMPORTANT: This must be called when a session is active (after session_start()).
+     * After calling this method, the browser will have a new session cookie,
+     * and other browsers using the old session ID will be logged out.
+     *
+     * @param string $newSessionId The target session ID to migrate to
+     * @param bool $deleteOldSession Whether to delete the old session data (default: true)
+     * @throws MigrationException If migration fails
+     * @throws InvalidSessionIdException If session ID is invalid
+     */
+    public function migrate(string $newSessionId, bool $deleteOldSession = true): void
```

PHPMD (2.15.0)

The method migrate has a boolean flag argument $deleteOldSession, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)